source/fundamentals/typescript.txt

```diff
@@ -122,6 +121,45 @@ The type error raised by the preceding code snippet is as follows:
    ...
    Type '"Sylvilagus"' is not assignable to type 'Condition<"Canis" | "Felis">'.
 
+Creating Keys from Variables
+````````````````````````````
+
+If you need to query a collection or perform another operation using a
+key created from specified variables, you must use an ``as const``
```

Comment:
Issue: Pretty sure I get this, but the wording is a little strange. When I think of a key being "created", I think of the operation that sets a value into a key that never existed (like your code sample below). Does that same thing happen when just performing a query, like you mention in the previous line? In other words, is the `as const` assertion required _any time_ you're referring to a key via a variable, or just when creating the key?

I think you can also drop the word "specified" before "variables; it's likely assumed that if the user is using a variable, they have control over it.

Suggestion: Specify the cases in which this is relevant

Reply:
By creating a key I just mean selecting one. The key used in the example may or may not exist -- the document is not given for us to confirm. I mean creating a key reference -- putting together a key (basically a path to a nested (it technically could be top level too) field name). `as const` is requires any time you refer to a key that incorporates a variable as a field/nested field/index/part of a field name/whatever else. I updated the subsection header and example introduction to hopefully make this more clear.